src/OpenLoco/Company.cpp

```diff
@@ -131,4 +132,44 @@ namespace OpenLoco
             unk.var_7C = totalRunCost;
         }
     }
+
+    // 0x004B8ED2
+    void Company::updateVehicleColours()
+    {
+        registers regs;
```

Comment:
Thought this was going to be a quick win implementation but turned out a little more complex than i thought.

```
for (auto v : VehicleList()){
    if (v->owner != id()){
        continue;
    }
    Vehicle train(v);
    for (auto& car : train.cars){
        auto* vehObject = car.firstCar.front->object();
        auto colour = mainColour;
        if (customVehicleColoursSet & (1 << vehObject->colour_type)){
            colour = vehicleColours[vehObject->colour_type - 1];
        }
        for (auto& carComponent : car){
            carComponent.front->colour_scheme = colour;
            carComponent.back->colour_scheme = colour;
            carComponent.body->colour_scheme = colour;
        }
    }
}
```